### PR TITLE
Fix learnergroup adhocgroup exam syncing

### LINKED ISSUE #13456: LOD - Quizzes assigned to individual learners are not being synced to the LOD device
## Observed behavior

When as a Coach I create a quiz which is assigned only to individual learners, then when I create a learn-only device and sign in as the learner, the quiz is not being synced to that device and the learner cannot see it.
Note that lessons assigned to individual learners are being synced correctly in this scenario.
Note also that this issue is valid only for learners on LODs as otherwise a quiz assigned to individual learners can be viewed and completed by the learners.
This issue is also present in Kolibri 0.18 and is not a regression caused by changes made in https://github.com/learningequality/kolibri/releases/tag/v0.18.1-beta0

https://github.com/user-attachments/assets/b06acc04-8f38-4c0e-af3e-f17b8d31683c

## Expected behavior

Quizzes assigned to individual learners should be synced correctly to a LOD so that a learner can view and complete them.

## Steps to reproduce the issue

1. Install Kolibri 0.18 or https://github.com/learningequality/kolibri/releases/tag/v0.18.1-beta0
2. Create a quiz and assign it to an individual learner
3. On another device setup a Learn-only device and sign in as that learner
4. Observe that the quiz is not being synced to the device

## Logs

[Logs-DB.zip](https://github.com/user-attachments/files/20611193/Logs-DB.zip)

## Usage Details
Kolibri 0.18 and Kolibri 0.18.1-beta0
Windows 11, Ubuntu 22  - Chrome, Firefox, Android

### PR DESCRIPTION
**Pull request by marcellamaki**
## Summary
Fixed single-user sync for exam assignments to LearnerGroups and AdHocGroups by fixing the exam assignment logic. Previously, the exam's collection_id was being set to the target collection (i.e. the group) rather than the classroom id, which broke the proper association. These changes ensure that the collection for the assignment object is set to the group, and the collection for the exam object is set to the classroom. 

## References
Fixes #13456

## Reviewer guidance
1. Create a quiz and assign to either a learner group, or a collection of individual learner (an adhoc group). These learners should be on LODs
2. Ensure that the quizzes properly sync to the LOD devices

### CODE REVIEW

**Comment by marcellamaki** [comment]
@coderabbitai full review

File: kolibri/core/exams/single_user_assignment_utils.py [hunk 1]
```diff
@@ -104,7 +110,13 @@ def update_assignments_from_individual_syncable_exams(user_id):
             or syncableexam.collection_id != assignment.collection_id
         ):
             exam = IndividualSyncableExam.deserialize_exam(syncableexam.serialized_exam)
-            exam.collection_id = syncableexam.collection_id
+            if syncableexam.collection.kind == collection_kinds.CLASSROOM:
```
Comment (rtibbles) [inline kolibri/core/exams/single_user_assignment_utils.py]: Maybe just abstract this logic into a method `_get_classroom_id_from_syncable_exam` and reuse in both places?

If we then wanted to remove the popping of `collection_id` from [here](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/exams/models.py#L451), we could then just do:
```
exam.collection_id = exam.collection_id or _get_classroom_id_from_syncable_exam(syncable_exam)
```

**Comment by marcellamaki** [comment]
@coderabbitai review

**Comment by marcellamaki** [comment]
@pcenov @radina - this is ready for manual QA. I have done preliminary manual QA myself to confirm the fix is working (one quiz assigned to one LOD learner individually, and it synced to the device) and Richard and I feel moderately confident in the fix and the regression tests that this should account for most cases. However, considering we missed this for.... a long time.... various combinations would help provide some peace of mind against edge cases. So things like testing: 
- the same exam not assigned to the whole class, but to multiple groups
- multiple individual learners


probably more, but i can't come up with any since it's late on friday. We can check in on Monday to discuss other scenarios if that's helpful. Thank you very much!

**Comment by pcenov** [comment]
Hi @marcellamaki - I confirm that now quizzes assigned to individual learners are being synced to the learn-only devices and there are no new issues observed while manually testing and regression testing today prior to [this commit](https://github.com/learningequality/kolibri/pull/13485/commits/3830f7ccd7c44da3bad92c51daa570e436b9f55c). :)

I tried multiple different scenarios: assigning and syncing quizzes to the entire class, individual learners, groups, combinations of groups and learners, multiple individual learners, multiple groups...  :)

While regression testing I noticed the following unrelated issue (extant also in 0.18): if at `Device > Settings` I have left unchecked the checkbox `Allow other devices on this network to view and import my unlisted channels` then all the assigned lessons and quizzes containing resources from the unlisted "QA Channel" are still being synced to the LODs. I double checked and this is not the case in Kolibri 0.17.5 where in that case the resources are not being synced to the LODs. So I am wondering whether this is an intentional change or I should file a follow-up issue for that?

**Comment by marcellamaki** [comment]
@pcenov (@radinamatic) - I've discussed with @rtibbles and it's _not_ an intentional change, but the first version was not ever intended behavior. So, we've addressed something that we didn't proactively choose to set up that way. We've concluded that we do want these resources to sync, because in this case, they're being intentionally assigned. Which is related to, but distinct from, browsing and importing from channels independently. 

I think as long as the browsing and importing behavior _separate_ from syncing is still not permitted, we can consider this the intended behavior.

**Review by radinamatic — APPROVED** [review]
Manual QA passed, good to go! 👍🏽 💯 :shipit:
